src/components/order-service/order-service.controller.ts

```diff
@@ -28,80 +40,114 @@ export class OrderServiceController implements angular.IController {
+    }, {
+      label: 'Bind',
+      id: 'bind',
+      view: 'order-service/order-service-bind.html',
+      hidden: false,
+      allowed: false,
+      valid: true,
+      onShow: this.showBind
     }, {
       label: 'Results',
       id: 'results',
-      view: 'order-service/order-service-review.html'
+      view: 'order-service/order-service-review.html',
+      hidden: false,
+      allowed: false,
+      valid: true,
+      prevEnabled: false,
+      onShow: this.showResults
     }];
-    if (this.ctrl.plans.length < 2) {
-      // Remove the plans step if there's only one plan.
-      this.ctrl.steps.shift();
-    }
-    this.gotoStep(this.ctrl.steps[0]);
     this.ctrl.nameTaken = false;
     this.ctrl.wizardReady = true;
-  }
+    this.ctrl.wizardDone = false;
+    this.ctrl.updating = false;
+
+    this.selectedProjectWatch = this.$scope.$watch(
+      () => {
+        return this.ctrl.selectedProject;
+      },
+      this.onProjectUpdate
+    );
 
-  public getSteps() {
-    return this.ctrl.steps;
+    var humanizeKind = this.$filter('humanizeKind');
+    this.ctrl.groupByKind = function(object: any) {
+      return humanizeKind(object.kind);
+    };
   }
 
-  public stepClick(step: any) {
-    // Prevent returning to previous steps if the order is complete.
-    if (this.ctrl.currentStep.id === 'results') {
-      return;
+  public showPageDefault = () => {
+    if (this.validityWatcher) {
+      this.validityWatcher();
+      this.validityWatcher = undefined;
     }
+    this.ctrl.nextTitle = 'Next >';
+    this.ctrl.steps[3].allowed = false;
+  };
 
-    if (!step.visited) {
-      return;
-    }
+  public showConfig = () => {
+    this.showPageDefault();
+    this.ctrl.steps[3].allowed = this.ctrl.steps[2].hidden && this.ctrl.steps[1].valid;
 
-    this.gotoStep(step);
-  }
+    this.validityWatcher = this.$scope.$watch("$ctrl.forms.orderConfigureForm.$valid", (isValid: any, lastValue: any) => {
+      this.ctrl.steps[1].valid = isValid;
+      this.ctrl.steps[2].allowed = this.ctrl.steps[1].valid;
+      this.ctrl.steps[3].allowed = this.ctrl.steps[2].hidden && this.ctrl.steps[1].valid;
+    });
+  };
 
-  public gotoStep(step: any) {
-    this.ctrl.steps.forEach((step: any) => step.selected = false);
-    if (this.ctrl.currentStep) {
-      this.ctrl.currentStep.visited = true;
-    }
-    this.ctrl.currentStep = step;
-    this.ctrl.currentStep.selected = true;
-    this.currentStepIndex = _.findIndex(this.ctrl.steps, 'selected');
-  }
+  public showBind = () => {
+    this.showPageDefault();
+    this.ctrl.nextTitle = 'Create';
+    this.ctrl.steps[3].allowed = this.ctrl.steps[2].valid;
 
-  public previousStep() {
-    let step = this.ctrl.steps[this.currentStepIndex - 1];
-    this.gotoStep(step);
-  }
+    this.validityWatcher = this.$scope.$watch("$ctrl.forms.bindForm.$valid", (isValid: any, lastValue: any) => {
+      this.ctrl.steps[2].valid = isValid;
+      this.ctrl.steps[3].allowed = this.ctrl.steps[2].valid;
```

Comment:
Agreed. The indexing got used more and more as I continued to develop and I didn't look back over it. Makes good sense, I will update.